// @syncfusion/ej2-vue-charts/src/stock-chart/axes.directive.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { EJComponentDecorator, allVue, gh } from '@syncfusion/ej2-vue-base';

import { isNullOrUndefined } from '@syncfusion/ej2-base';
import Vue from 'vue';

export var isExecute = gh ? false : true;
var vueImport;
if (!isExecute || parseInt(allVue.version) < 3) {
    vueImport = Vue;
}
else {
    vueImport = Vue;
}
var StockChartAxesDirective = /** @class */ (function (_super) {
    __extends(StockChartAxesDirective, _super);
    function StockChartAxesDirective() {
        return _super.call(this, arguments) || this;
    }
    StockChartAxesDirective.prototype.render = function (createElement) {
        if (gh) {
            var h = gh || createElement;
            var slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    };
    StockChartAxesDirective.prototype.updated = function () {
        if (gh && this.custom) {
            this.custom();
        }
    };
    StockChartAxesDirective.prototype.getTag = function () {
        return 'e-stockchart-axes';
    };
    StockChartAxesDirective = __decorate([
        EJComponentDecorator({}, isExecute)
        
    ], StockChartAxesDirective);
    return StockChartAxesDirective;
}(vueImport));
export { StockChartAxesDirective };
export var StockChartAxesPlugin = {
    name: 'e-stockchart-axes',
    install: function (Vue) {
        Vue.component(StockChartAxesPlugin.name, StockChartAxesDirective);
    }
};
var StockChartAxisDirective = /** @class */ (function (_super) {
    __extends(StockChartAxisDirective, _super);
    function StockChartAxisDirective() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    StockChartAxisDirective.prototype.render = function () {
        return;
    };
    StockChartAxisDirective.prototype.getTag = function () {
        return 'e-stockchart-axis';
    };
    StockChartAxisDirective = __decorate([
        EJComponentDecorator({}, isExecute)
    ], StockChartAxisDirective);
    return StockChartAxisDirective;
}(vueImport));
export { StockChartAxisDirective };
export var StockChartAxisPlugin = {
    name: 'e-stockchart-axis',
    install: function (Vue) {
        Vue.component(StockChartAxisPlugin.name, StockChartAxisDirective);
    }
};
