// @syncfusion/ej2-vue-charts/dist/es6/ej2-vue-charts.es2015.js
import { AccumulationChart, BulletChart, Chart, RangeNavigator, Smithchart, Sparkline, StockChart } from '@syncfusion/ej2-charts';
import { ComponentBase, EJComponentDecorator, allVue, getProps, gh } from '@syncfusion/ej2-vue-base';
import { Vue } from 'vue-class-component';
import { getValue, isNullOrUndefined, isUndefined } from '@syncfusion/ej2-base';
import Vue$1 from 'vue';

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute = gh ? false : true;
let vueImport;
if (!isExecute || parseInt(allVue.version) < 3) {
    vueImport = Vue;
}
else {
    vueImport = Vue$1;
}
let TrendlinesDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class TrendlinesDirective extends vueImport {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-trendlines';
    }
};
TrendlinesDirective = __decorate([
    EJComponentDecorator({}, isExecute)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], TrendlinesDirective);
const TrendlinesPlugin = {
    name: 'e-trendlines',
    install(Vue$$1) {
        Vue$$1.component(TrendlinesPlugin.name, TrendlinesDirective);
    }
};
let TrendlineDirective = class TrendlineDirective extends vueImport {
    render() {
        return;
    }
    getTag() {
        return 'e-trendline';
    }
};
TrendlineDirective = __decorate([
    EJComponentDecorator({}, isExecute)
], TrendlineDirective);
const TrendlinePlugin = {
    name: 'e-trendline',
    install(Vue$$1) {
        Vue$$1.component(TrendlinePlugin.name, TrendlineDirective);
    }
};

var __decorate$1 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$1 = gh ? false : true;
let vueImport$1;
if (!isExecute$1 || parseInt(allVue.version) < 3) {
    vueImport$1 = Vue;
}
else {
    vueImport$1 = Vue$1;
}
let SegmentsDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class SegmentsDirective extends vueImport$1 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-segments';
    }
};
SegmentsDirective = __decorate$1([
    EJComponentDecorator({}, isExecute$1)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], SegmentsDirective);
const SegmentsPlugin = {
    name: 'e-segments',
    install(Vue$$1) {
        Vue$$1.component(SegmentsPlugin.name, SegmentsDirective);
    }
};
let SegmentDirective = class SegmentDirective extends vueImport$1 {
    render() {
        return;
    }
    getTag() {
        return 'e-segment';
    }
};
SegmentDirective = __decorate$1([
    EJComponentDecorator({}, isExecute$1)
], SegmentDirective);
const SegmentPlugin = {
    name: 'e-segment',
    install(Vue$$1) {
        Vue$$1.component(SegmentPlugin.name, SegmentDirective);
    }
};

var __decorate$2 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$2 = gh ? false : true;
let vueImport$2;
if (!isExecute$2 || parseInt(allVue.version) < 3) {
    vueImport$2 = Vue;
}
else {
    vueImport$2 = Vue$1;
}
let SeriesCollectionDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class SeriesCollectionDirective extends vueImport$2 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-series-collection';
    }
};
SeriesCollectionDirective = __decorate$2([
    EJComponentDecorator({}, isExecute$2)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], SeriesCollectionDirective);
const SeriesCollectionPlugin = {
    name: 'e-series-collection',
    install(Vue$$1) {
        Vue$$1.component(SeriesCollectionPlugin.name, SeriesCollectionDirective);
    }
};
let SeriesDirective = class SeriesDirective extends vueImport$2 {
    render() {
        return;
    }
    getTag() {
        return 'e-series';
    }
};
SeriesDirective = __decorate$2([
    EJComponentDecorator({}, isExecute$2)
], SeriesDirective);
const SeriesPlugin = {
    name: 'e-series',
    install(Vue$$1) {
        Vue$$1.component(SeriesPlugin.name, SeriesDirective);
    }
};

var __decorate$3 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$3 = gh ? false : true;
let vueImport$3;
if (!isExecute$3 || parseInt(allVue.version) < 3) {
    vueImport$3 = Vue;
}
else {
    vueImport$3 = Vue$1;
}
let StripLinesDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class StripLinesDirective extends vueImport$3 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-striplines';
    }
};
StripLinesDirective = __decorate$3([
    EJComponentDecorator({}, isExecute$3)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], StripLinesDirective);
const StripLinesPlugin = {
    name: 'e-striplines',
    install(Vue$$1) {
        Vue$$1.component(StripLinesPlugin.name, StripLinesDirective);
    }
};
let StripLineDirective = class StripLineDirective extends vueImport$3 {
    render() {
        return;
    }
    getTag() {
        return 'e-stripline';
    }
};
StripLineDirective = __decorate$3([
    EJComponentDecorator({}, isExecute$3)
], StripLineDirective);
const StripLinePlugin = {
    name: 'e-stripline',
    install(Vue$$1) {
        Vue$$1.component(StripLinePlugin.name, StripLineDirective);
    }
};

var __decorate$4 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$4 = gh ? false : true;
let vueImport$4;
if (!isExecute$4 || parseInt(allVue.version) < 3) {
    vueImport$4 = Vue;
}
else {
    vueImport$4 = Vue$1;
}
let CategoriesDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class CategoriesDirective extends vueImport$4 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-categories';
    }
};
CategoriesDirective = __decorate$4([
    EJComponentDecorator({}, isExecute$4)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], CategoriesDirective);
const CategoriesPlugin = {
    name: 'e-categories',
    install(Vue$$1) {
        Vue$$1.component(CategoriesPlugin.name, CategoriesDirective);
    }
};
let CategoryDirective = class CategoryDirective extends vueImport$4 {
    render() {
        return;
    }
    getTag() {
        return 'e-category';
    }
};
CategoryDirective = __decorate$4([
    EJComponentDecorator({}, isExecute$4)
], CategoryDirective);
const CategoryPlugin = {
    name: 'e-category',
    install(Vue$$1) {
        Vue$$1.component(CategoryPlugin.name, CategoryDirective);
    }
};

var __decorate$5 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$5 = gh ? false : true;
let vueImport$5;
if (!isExecute$5 || parseInt(allVue.version) < 3) {
    vueImport$5 = Vue;
}
else {
    vueImport$5 = Vue$1;
}
let MultiLevelLabelsDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class MultiLevelLabelsDirective extends vueImport$5 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-multilevellabels';
    }
};
MultiLevelLabelsDirective = __decorate$5([
    EJComponentDecorator({}, isExecute$5)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], MultiLevelLabelsDirective);
const MultiLevelLabelsPlugin = {
    name: 'e-multilevellabels',
    install(Vue$$1) {
        Vue$$1.component(MultiLevelLabelsPlugin.name, MultiLevelLabelsDirective);
    }
};
let MultiLevelLabelDirective = class MultiLevelLabelDirective extends vueImport$5 {
    render() {
        return;
    }
    getTag() {
        return 'e-multilevellabel';
    }
};
MultiLevelLabelDirective = __decorate$5([
    EJComponentDecorator({}, isExecute$5)
], MultiLevelLabelDirective);
const MultiLevelLabelPlugin = {
    name: 'e-multilevellabel',
    install(Vue$$1) {
        Vue$$1.component(MultiLevelLabelPlugin.name, MultiLevelLabelDirective);
    }
};

var __decorate$6 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$6 = gh ? false : true;
let vueImport$6;
if (!isExecute$6 || parseInt(allVue.version) < 3) {
    vueImport$6 = Vue;
}
else {
    vueImport$6 = Vue$1;
}
let AxesDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class AxesDirective extends vueImport$6 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-axes';
    }
};
AxesDirective = __decorate$6([
    EJComponentDecorator({}, isExecute$6)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], AxesDirective);
const AxesPlugin = {
    name: 'e-axes',
    install(Vue$$1) {
        Vue$$1.component(AxesPlugin.name, AxesDirective);
    }
};
let AxisDirective = class AxisDirective extends vueImport$6 {
    render() {
        return;
    }
    getTag() {
        return 'e-axis';
    }
};
AxisDirective = __decorate$6([
    EJComponentDecorator({}, isExecute$6)
], AxisDirective);
const AxisPlugin = {
    name: 'e-axis',
    install(Vue$$1) {
        Vue$$1.component(AxisPlugin.name, AxisDirective);
    }
};

var __decorate$7 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$7 = gh ? false : true;
let vueImport$7;
if (!isExecute$7 || parseInt(allVue.version) < 3) {
    vueImport$7 = Vue;
}
else {
    vueImport$7 = Vue$1;
}
let RowsDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class RowsDirective extends vueImport$7 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-rows';
    }
};
RowsDirective = __decorate$7([
    EJComponentDecorator({}, isExecute$7)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], RowsDirective);
const RowsPlugin = {
    name: 'e-rows',
    install(Vue$$1) {
        Vue$$1.component(RowsPlugin.name, RowsDirective);
    }
};
let RowDirective = class RowDirective extends vueImport$7 {
    render() {
        return;
    }
    getTag() {
        return 'e-row';
    }
};
RowDirective = __decorate$7([
    EJComponentDecorator({}, isExecute$7)
], RowDirective);
const RowPlugin = {
    name: 'e-row',
    install(Vue$$1) {
        Vue$$1.component(RowPlugin.name, RowDirective);
    }
};

var __decorate$8 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$8 = gh ? false : true;
let vueImport$8;
if (!isExecute$8 || parseInt(allVue.version) < 3) {
    vueImport$8 = Vue;
}
else {
    vueImport$8 = Vue$1;
}
let ColumnsDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class ColumnsDirective extends vueImport$8 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-columns';
    }
};
ColumnsDirective = __decorate$8([
    EJComponentDecorator({}, isExecute$8)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], ColumnsDirective);
const ColumnsPlugin = {
    name: 'e-columns',
    install(Vue$$1) {
        Vue$$1.component(ColumnsPlugin.name, ColumnsDirective);
    }
};
let ColumnDirective = class ColumnDirective extends vueImport$8 {
    render() {
        return;
    }
    getTag() {
        return 'e-column';
    }
};
ColumnDirective = __decorate$8([
    EJComponentDecorator({}, isExecute$8)
], ColumnDirective);
const ColumnPlugin = {
    name: 'e-column',
    install(Vue$$1) {
        Vue$$1.component(ColumnPlugin.name, ColumnDirective);
    }
};

var __decorate$9 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$9 = gh ? false : true;
let vueImport$9;
if (!isExecute$9 || parseInt(allVue.version) < 3) {
    vueImport$9 = Vue;
}
else {
    vueImport$9 = Vue$1;
}
let RangeColorSettingsDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class RangeColorSettingsDirective extends vueImport$9 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-rangecolorsettings';
    }
};
RangeColorSettingsDirective = __decorate$9([
    EJComponentDecorator({}, isExecute$9)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], RangeColorSettingsDirective);
const RangeColorSettingsPlugin = {
    name: 'e-rangecolorsettings',
    install(Vue$$1) {
        Vue$$1.component(RangeColorSettingsPlugin.name, RangeColorSettingsDirective);
    }
};
let RangeColorSettingDirective = class RangeColorSettingDirective extends vueImport$9 {
    render() {
        return;
    }
    getTag() {
        return 'e-rangecolorsetting';
    }
};
RangeColorSettingDirective = __decorate$9([
    EJComponentDecorator({}, isExecute$9)
], RangeColorSettingDirective);
const RangeColorSettingPlugin = {
    name: 'e-rangecolorsetting',
    install(Vue$$1) {
        Vue$$1.component(RangeColorSettingPlugin.name, RangeColorSettingDirective);
    }
};

var __decorate$10 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$10 = gh ? false : true;
let vueImport$10;
if (!isExecute$10 || parseInt(allVue.version) < 3) {
    vueImport$10 = Vue;
}
else {
    vueImport$10 = Vue$1;
}
let AnnotationsDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class AnnotationsDirective extends vueImport$10 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-annotations';
    }
};
AnnotationsDirective = __decorate$10([
    EJComponentDecorator({}, isExecute$10)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], AnnotationsDirective);
const AnnotationsPlugin = {
    name: 'e-annotations',
    install(Vue$$1) {
        Vue$$1.component(AnnotationsPlugin.name, AnnotationsDirective);
    }
};
/**
 * `e-annotation` directive represent a annotation of the VueJS Chart.
 * It must be contained in a Chart component(`ejs-chart`).
 * ```vue
 * <ejs-chart>
 *   <e-annotations>
 *    <e-annotation content='ID' />
 *    <e-annotation content='ID' />
 *   </e-annotations>
 * </ejs-chart>
 * ```
 */
let AnnotationDirective = class AnnotationDirective extends vueImport$10 {
    render() {
        return;
    }
    getTag() {
        return 'e-annotation';
    }
};
AnnotationDirective = __decorate$10([
    EJComponentDecorator({}, isExecute$10)
], AnnotationDirective);
const AnnotationPlugin = {
    name: 'e-annotation',
    install(Vue$$1) {
        Vue$$1.component(AnnotationPlugin.name, AnnotationDirective);
    }
};

var __decorate$11 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$11 = gh ? false : true;
let vueImport$11;
if (!isExecute$11 || parseInt(allVue.version) < 3) {
    vueImport$11 = Vue;
}
else {
    vueImport$11 = Vue$1;
}
let SelectedDataIndexesDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class SelectedDataIndexesDirective extends vueImport$11 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-selecteddataindexes';
    }
};
SelectedDataIndexesDirective = __decorate$11([
    EJComponentDecorator({}, isExecute$11)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], SelectedDataIndexesDirective);
const SelectedDataIndexesPlugin = {
    name: 'e-selecteddataindexes',
    install(Vue$$1) {
        Vue$$1.component(SelectedDataIndexesPlugin.name, SelectedDataIndexesDirective);
    }
};
let SelectedDataIndexDirective = class SelectedDataIndexDirective extends vueImport$11 {
    render() {
        return;
    }
    getTag() {
        return 'e-selecteddataindex';
    }
};
SelectedDataIndexDirective = __decorate$11([
    EJComponentDecorator({}, isExecute$11)
], SelectedDataIndexDirective);
const SelectedDataIndexPlugin = {
    name: 'e-selecteddataindex',
    install(Vue$$1) {
        Vue$$1.component(SelectedDataIndexPlugin.name, SelectedDataIndexDirective);
    }
};

var __decorate$12 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$12 = gh ? false : true;
let vueImport$12;
if (!isExecute$12 || parseInt(allVue.version) < 3) {
    vueImport$12 = Vue;
}
else {
    vueImport$12 = Vue$1;
}
let IndicatorsDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class IndicatorsDirective extends vueImport$12 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-indicators';
    }
};
IndicatorsDirective = __decorate$12([
    EJComponentDecorator({}, isExecute$12)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], IndicatorsDirective);
const IndicatorsPlugin = {
    name: 'e-indicators',
    install(Vue$$1) {
        Vue$$1.component(IndicatorsPlugin.name, IndicatorsDirective);
    }
};
let IndicatorDirective = class IndicatorDirective extends vueImport$12 {
    render() {
        return;
    }
    getTag() {
        return 'e-indicator';
    }
};
IndicatorDirective = __decorate$12([
    EJComponentDecorator({}, isExecute$12)
], IndicatorDirective);
const IndicatorPlugin = {
    name: 'e-indicator',
    install(Vue$$1) {
        Vue$$1.component(IndicatorPlugin.name, IndicatorDirective);
    }
};

var __decorate$13 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const properties = ['isLazyUpdate', 'allowExport', 'allowMultiSelection', 'annotations', 'axes', 'background', 'backgroundImage', 'border', 'chartArea', 'columns', 'crosshair', 'currencyCode', 'dataSource', 'description', 'enableAnimation', 'enableAutoIntervalOnBothAxis', 'enableCanvas', 'enableExport', 'enablePersistence', 'enableRtl', 'enableSideBySidePlacement', 'height', 'highlightColor', 'highlightMode', 'highlightPattern', 'indicators', 'isMultiSelect', 'isTransposed', 'legendSettings', 'locale', 'margin', 'palettes', 'primaryXAxis', 'primaryYAxis', 'rangeColorSettings', 'rows', 'selectedDataIndexes', 'selectionMode', 'selectionPattern', 'series', 'subTitle', 'subTitleStyle', 'tabIndex', 'theme', 'title', 'titleStyle', 'tooltip', 'useGroupingSeparator', 'width', 'zoomSettings', 'afterExport', 'animationComplete', 'annotationRender', 'axisLabelClick', 'axisLabelRender', 'axisMultiLabelRender', 'axisRangeCalculated', 'beforeExport', 'beforePrint', 'beforeResize', 'chartDoubleClick', 'chartMouseClick', 'chartMouseDown', 'chartMouseLeave', 'chartMouseMove', 'chartMouseUp', 'drag', 'dragComplete', 'dragEnd', 'dragStart', 'legendClick', 'legendRender', 'load', 'loaded', 'multiLevelLabelClick', 'onZooming', 'pointClick', 'pointDoubleClick', 'pointMove', 'pointRender', 'resized', 'scrollChanged', 'scrollEnd', 'scrollStart', 'selectionComplete', 'seriesRender', 'sharedTooltipRender', 'textRender', 'tooltipRender', 'zoomComplete'];
const modelProps = ['dataSource'];
const testProp = getProps({ props: properties });
const props = testProp[0];
const watch = testProp[1];
const emitProbs = Object.keys(watch);
emitProbs.push('modelchanged');
for (let props of modelProps) {
    emitProbs.push('update:' + props);
}
const isExecute$13 = gh ? false : true;
/**
 * Represents Vuejs chart Component
 * ```vue
 * <ejs-chart></ejs-chart>
 * ```
 */
let ChartComponent = 
/* Start Options({
    props: props,
    watch: watch,
    emits: emitProbs,
    provide: function provide() {
        return {
            custom: this.custom
        };
    }
}) End */
class ChartComponent extends ComponentBase {
    constructor() {
        super(arguments);
        this.propKeys = properties;
        this.models = modelProps;
        this.hasChildDirective = true;
        this.hasInjectedModules = true;
        this.tagMapper = { "e-series-collection": { "e-series": { "e-trendlines": "e-trendline", "e-segments": "e-segment" } }, "e-axes": { "e-axis": { "e-striplines": "e-stripline", "e-multilevellabels": { "e-multilevellabel": { "e-categories": "e-category" } } } }, "e-rows": "e-row", "e-columns": "e-column", "e-rangecolorsettings": "e-rangecolorsetting", "e-annotations": "e-annotation", "e-selecteddataindexes": "e-selecteddataindex", "e-indicators": "e-indicator" };
        this.tagNameMapper = { "e-series-collection": "e-series", "e-striplines": "e-stripLines", "e-multilevellabels": "e-multiLevelLabels", "e-rangecolorsettings": "e-rangeColorSettings", "e-selecteddataindexes": "e-selectedDataIndexes" };
        this.isVue3 = !isExecute$13;
        this.ej2Instances = new Chart({});
        this.ej2Instances._trigger = this.ej2Instances.trigger;
        this.ej2Instances.trigger = this.trigger;
        this.bindProperties();
        this.ej2Instances._setProperties = this.ej2Instances.setProperties;
        this.ej2Instances.setProperties = this.setProperties;
        this.ej2Instances.clearTemplate = this.clearTemplate;
        this.updated = this.updated;
    }
    clearTemplate(templateNames) {
        if (!templateNames) {
            templateNames = Object.keys(this.templateCollection || {});
        }
        if (templateNames.length && this.templateCollection) {
            for (let tempName of templateNames) {
                let elementCollection = this.templateCollection[tempName];
                if (elementCollection && elementCollection.length) {
                    for (let ele of elementCollection) {
                        let destroy = getValue('__vue__.$destroy', ele);
                        if (destroy) {
                            ele.__vue__.$destroy();
                        }
                        if (ele.innerHTML) {
                            ele.innerHTML = '';
                        }
                    }
                    delete this.templateCollection[tempName];
                }
            }
        }
    }
    setProperties(prop, muteOnChange) {
        if (this.isVue3) {
            this.models = !this.models ? this.ej2Instances.referModels : this.models;
        }
        if (this.ej2Instances && this.ej2Instances._setProperties) {
            this.ej2Instances._setProperties(prop, muteOnChange);
        }
        if (prop && this.models && this.models.length) {
            Object.keys(prop).map((key) => {
                this.models.map((model) => {
                    if ((key === model) && !(/datasource/i.test(key))) {
                        if (this.isVue3) {
                            this.ej2Instances.vueInstance.$emit('update:' + key, prop[key]);
                        }
                        else {
                            this.$emit('update:' + key, prop[key]);
                            this.$emit('modelchanged', prop[key]);
                        }
                    }
                });
            });
        }
    }
    trigger(eventName, eventProp, successHandler) {
        if (!isExecute$13) {
            this.models = !this.models ? this.ej2Instances.referModels : this.models;
        }
        if ((eventName === 'change' || eventName === 'input') && this.models && (this.models.length !== 0)) {
            let key = this.models.toString().match(/checked|value/) || [];
            let propKey = key[0];
            if (eventProp && key && !isUndefined(eventProp[propKey])) {
                if (!isExecute$13) {
                    this.ej2Instances.vueInstance.$emit('update:' + propKey, eventProp[propKey]);
                    this.ej2Instances.vueInstance.$emit('modelchanged', eventProp[propKey]);
                }
                else {
                    if (eventName === 'change' || (this.$props && !this.$props.isLazyUpdate)) {
                        this.$emit('update:' + propKey, eventProp[propKey]);
                        this.$emit('modelchanged', eventProp[propKey]);
                    }
                }
            }
        }
        else if ((eventName === 'actionBegin' && eventProp.requestType === 'dateNavigate') && this.models && (this.models.length !== 0)) {
            let key = this.models.toString().match(/currentView|selectedDate/) || [];
            let propKey = key[0];
            if (eventProp && key && !isUndefined(eventProp[propKey])) {
                if (!isExecute$13) {
                    this.ej2Instances.vueInstance.$emit('update:' + propKey, eventProp[propKey]);
                    this.ej2Instances.vueInstance.$emit('modelchanged', eventProp[propKey]);
                }
                else {
                    this.$emit('update:' + propKey, eventProp[propKey]);
                    this.$emit('modelchanged', eventProp[propKey]);
                }
            }
        }
        if ((this.ej2Instances && this.ej2Instances._trigger)) {
            this.ej2Instances._trigger(eventName, eventProp, successHandler);
        }
    }
    render(createElement) {
        let h = gh || createElement;
        let slots = null;
        if (!isNullOrUndefined(this.$slots.default)) {
            slots = gh ? this.$slots.default() : this.$slots.default;
        }
        return h('div', slots);
    }
    custom() {
        this.updated();
    }
    addAxes(axisCollection) {
        return this.ej2Instances.addAxes(axisCollection);
    }
    addSeries(seriesCollection) {
        return this.ej2Instances.addSeries(seriesCollection);
    }
    clearSeries() {
        return this.ej2Instances.clearSeries();
    }
    createChartSvg() {
        return this.ej2Instances.createChartSvg();
    }
    export(type, fileName) {
        return this.ej2Instances.export(type, fileName);
    }
    getLocalizedLabel(key) {
        return this.ej2Instances.getLocalizedLabel(key);
    }
    print(id) {
        return this.ej2Instances.print(id);
    }
    processData(render) {
        return this.ej2Instances.processData(render);
    }
    refreshLiveData() {
        return this.ej2Instances.refreshLiveData();
    }
    removeAxis(index) {
        return this.ej2Instances.removeAxis(index);
    }
    removeSeries(index) {
        return this.ej2Instances.removeSeries(index);
    }
    setAnnotationValue(annotationIndex, content) {
        return this.ej2Instances.setAnnotationValue(annotationIndex, content);
    }
};
ChartComponent = __decorate$13([
    EJComponentDecorator({
        props: properties,
        model: {
            event: 'modelchanged'
        }
    }, isExecute$13)
    /* Start Options({
        props: props,
        watch: watch,
        emits: emitProbs,
        provide: function provide() {
            return {
                custom: this.custom
            };
        }
    }) End */
], ChartComponent);
const ChartPlugin = {
    name: 'ejs-chart',
    install(Vue$$1) {
        Vue$$1.component(ChartPlugin.name, ChartComponent);
        Vue$$1.component(SeriesPlugin.name, SeriesDirective);
        Vue$$1.component(SeriesCollectionPlugin.name, SeriesCollectionDirective);
        Vue$$1.component(TrendlinePlugin.name, TrendlineDirective);
        Vue$$1.component(TrendlinesPlugin.name, TrendlinesDirective);
        Vue$$1.component(SegmentPlugin.name, SegmentDirective);
        Vue$$1.component(SegmentsPlugin.name, SegmentsDirective);
        Vue$$1.component(AxisPlugin.name, AxisDirective);
        Vue$$1.component(AxesPlugin.name, AxesDirective);
        Vue$$1.component(StripLinePlugin.name, StripLineDirective);
        Vue$$1.component(StripLinesPlugin.name, StripLinesDirective);
        Vue$$1.component(MultiLevelLabelPlugin.name, MultiLevelLabelDirective);
        Vue$$1.component(MultiLevelLabelsPlugin.name, MultiLevelLabelsDirective);
        Vue$$1.component(CategoryPlugin.name, CategoryDirective);
        Vue$$1.component(CategoriesPlugin.name, CategoriesDirective);
        Vue$$1.component(RowPlugin.name, RowDirective);
        Vue$$1.component(RowsPlugin.name, RowsDirective);
        Vue$$1.component(ColumnPlugin.name, ColumnDirective);
        Vue$$1.component(ColumnsPlugin.name, ColumnsDirective);
        Vue$$1.component(RangeColorSettingPlugin.name, RangeColorSettingDirective);
        Vue$$1.component(RangeColorSettingsPlugin.name, RangeColorSettingsDirective);
        Vue$$1.component(AnnotationPlugin.name, AnnotationDirective);
        Vue$$1.component(AnnotationsPlugin.name, AnnotationsDirective);
        Vue$$1.component(SelectedDataIndexPlugin.name, SelectedDataIndexDirective);
        Vue$$1.component(SelectedDataIndexesPlugin.name, SelectedDataIndexesDirective);
        Vue$$1.component(IndicatorPlugin.name, IndicatorDirective);
        Vue$$1.component(IndicatorsPlugin.name, IndicatorsDirective);
    }
};

var __decorate$14 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$14 = gh ? false : true;
let vueImport$13;
if (!isExecute$14 || parseInt(allVue.version) < 3) {
    vueImport$13 = Vue;
}
else {
    vueImport$13 = Vue$1;
}
let AccumulationSeriesCollectionDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class AccumulationSeriesCollectionDirective extends vueImport$13 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-accumulation-series-collection';
    }
};
AccumulationSeriesCollectionDirective = __decorate$14([
    EJComponentDecorator({}, isExecute$14)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], AccumulationSeriesCollectionDirective);
const AccumulationSeriesCollectionPlugin = {
    name: 'e-accumulation-series-collection',
    install(Vue$$1) {
        Vue$$1.component(AccumulationSeriesCollectionPlugin.name, AccumulationSeriesCollectionDirective);
    }
};
let AccumulationSeriesDirective = class AccumulationSeriesDirective extends vueImport$13 {
    render() {
        return;
    }
    getTag() {
        return 'e-accumulation-series';
    }
};
AccumulationSeriesDirective = __decorate$14([
    EJComponentDecorator({}, isExecute$14)
], AccumulationSeriesDirective);
const AccumulationSeriesPlugin = {
    name: 'e-accumulation-series',
    install(Vue$$1) {
        Vue$$1.component(AccumulationSeriesPlugin.name, AccumulationSeriesDirective);
    }
};

var __decorate$15 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$15 = gh ? false : true;
let vueImport$14;
if (!isExecute$15 || parseInt(allVue.version) < 3) {
    vueImport$14 = Vue;
}
else {
    vueImport$14 = Vue$1;
}
let AccumulationAnnotationsDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class AccumulationAnnotationsDirective extends vueImport$14 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-accumulation-annotations';
    }
};
AccumulationAnnotationsDirective = __decorate$15([
    EJComponentDecorator({}, isExecute$15)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], AccumulationAnnotationsDirective);
const AccumulationAnnotationsPlugin = {
    name: 'e-accumulation-annotations',
    install(Vue$$1) {
        Vue$$1.component(AccumulationAnnotationsPlugin.name, AccumulationAnnotationsDirective);
    }
};
let AccumulationAnnotationDirective = class AccumulationAnnotationDirective extends vueImport$14 {
    render() {
        return;
    }
    getTag() {
        return 'e-accumulation-annotation';
    }
};
AccumulationAnnotationDirective = __decorate$15([
    EJComponentDecorator({}, isExecute$15)
], AccumulationAnnotationDirective);
const AccumulationAnnotationPlugin = {
    name: 'e-accumulation-annotation',
    install(Vue$$1) {
        Vue$$1.component(AccumulationAnnotationPlugin.name, AccumulationAnnotationDirective);
    }
};

var __decorate$16 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const properties$1 = ['isLazyUpdate', 'allowExport', 'annotations', 'background', 'backgroundImage', 'border', 'center', 'currencyCode', 'dataSource', 'enableAnimation', 'enableBorderOnMouseMove', 'enableExport', 'enablePersistence', 'enableRtl', 'enableSmartLabels', 'height', 'highlightMode', 'highlightPattern', 'isMultiSelect', 'legendSettings', 'locale', 'margin', 'selectedDataIndexes', 'selectionMode', 'selectionPattern', 'series', 'subTitle', 'subTitleStyle', 'theme', 'title', 'titleStyle', 'tooltip', 'useGroupingSeparator', 'width', 'afterExport', 'animationComplete', 'annotationRender', 'beforePrint', 'beforeResize', 'chartDoubleClick', 'chartMouseClick', 'chartMouseDown', 'chartMouseLeave', 'chartMouseMove', 'chartMouseUp', 'legendRender', 'load', 'loaded', 'pointClick', 'pointMove', 'pointRender', 'resized', 'selectionComplete', 'seriesRender', 'textRender', 'tooltipRender'];
const modelProps$1 = ['dataSource'];
const testProp$1 = getProps({ props: properties$1 });
const props$1 = testProp$1[0];
const watch$1 = testProp$1[1];
const emitProbs$1 = Object.keys(watch$1);
emitProbs$1.push('modelchanged');
for (let props of modelProps$1) {
    emitProbs$1.push('update:' + props);
}
const isExecute$16 = gh ? false : true;
/**
 * Represents Vuejs AccumulationChart Component
 * ```vue
 * <ejs-accumulationchart></ejs-accumulationchart>
 * ```
 */
let AccumulationChartComponent = 
/* Start Options({
    props: props,
    watch: watch,
    emits: emitProbs,
    provide: function provide() {
        return {
            custom: this.custom
        };
    }
}) End */
class AccumulationChartComponent extends ComponentBase {
    constructor() {
        super(arguments);
        this.propKeys = properties$1;
        this.models = modelProps$1;
        this.hasChildDirective = true;
        this.hasInjectedModules = true;
        this.tagMapper = { "e-accumulation-series-collection": "e-accumulation-series", "e-accumulation-annotations": "e-accumulation-annotation" };
        this.tagNameMapper = { "e-accumulation-series-collection": "e-series", "e-accumulation-annotations": "e-annotations" };
        this.isVue3 = !isExecute$16;
        this.ej2Instances = new AccumulationChart({});
        this.ej2Instances._trigger = this.ej2Instances.trigger;
        this.ej2Instances.trigger = this.trigger;
        this.bindProperties();
        this.ej2Instances._setProperties = this.ej2Instances.setProperties;
        this.ej2Instances.setProperties = this.setProperties;
        this.ej2Instances.clearTemplate = this.clearTemplate;
        this.updated = this.updated;
    }
    clearTemplate(templateNames) {
        if (!templateNames) {
            templateNames = Object.keys(this.templateCollection || {});
        }
        if (templateNames.length && this.templateCollection) {
            for (let tempName of templateNames) {
                let elementCollection = this.templateCollection[tempName];
                if (elementCollection && elementCollection.length) {
                    for (let ele of elementCollection) {
                        let destroy = getValue('__vue__.$destroy', ele);
                        if (destroy) {
                            ele.__vue__.$destroy();
                        }
                        if (ele.innerHTML) {
                            ele.innerHTML = '';
                        }
                    }
                    delete this.templateCollection[tempName];
                }
            }
        }
    }
    setProperties(prop, muteOnChange) {
        if (this.isVue3) {
            this.models = !this.models ? this.ej2Instances.referModels : this.models;
        }
        if (this.ej2Instances && this.ej2Instances._setProperties) {
            this.ej2Instances._setProperties(prop, muteOnChange);
        }
        if (prop && this.models && this.models.length) {
            Object.keys(prop).map((key) => {
                this.models.map((model) => {
                    if ((key === model) && !(/datasource/i.test(key))) {
                        if (this.isVue3) {
                            this.ej2Instances.vueInstance.$emit('update:' + key, prop[key]);
                        }
                        else {
                            this.$emit('update:' + key, prop[key]);
                            this.$emit('modelchanged', prop[key]);
                        }
                    }
                });
            });
        }
    }
    trigger(eventName, eventProp, successHandler) {
        if (!isExecute$16) {
            this.models = !this.models ? this.ej2Instances.referModels : this.models;
        }
        if ((eventName === 'change' || eventName === 'input') && this.models && (this.models.length !== 0)) {
            let key = this.models.toString().match(/checked|value/) || [];
            let propKey = key[0];
            if (eventProp && key && !isUndefined(eventProp[propKey])) {
                if (!isExecute$16) {
                    this.ej2Instances.vueInstance.$emit('update:' + propKey, eventProp[propKey]);
                    this.ej2Instances.vueInstance.$emit('modelchanged', eventProp[propKey]);
                }
                else {
                    if (eventName === 'change' || (this.$props && !this.$props.isLazyUpdate)) {
                        this.$emit('update:' + propKey, eventProp[propKey]);
                        this.$emit('modelchanged', eventProp[propKey]);
                    }
                }
            }
        }
        else if ((eventName === 'actionBegin' && eventProp.requestType === 'dateNavigate') && this.models && (this.models.length !== 0)) {
            let key = this.models.toString().match(/currentView|selectedDate/) || [];
            let propKey = key[0];
            if (eventProp && key && !isUndefined(eventProp[propKey])) {
                if (!isExecute$16) {
                    this.ej2Instances.vueInstance.$emit('update:' + propKey, eventProp[propKey]);
                    this.ej2Instances.vueInstance.$emit('modelchanged', eventProp[propKey]);
                }
                else {
                    this.$emit('update:' + propKey, eventProp[propKey]);
                    this.$emit('modelchanged', eventProp[propKey]);
                }
            }
        }
        if ((this.ej2Instances && this.ej2Instances._trigger)) {
            this.ej2Instances._trigger(eventName, eventProp, successHandler);
        }
    }
    render(createElement) {
        let h = gh || createElement;
        let slots = null;
        if (!isNullOrUndefined(this.$slots.default)) {
            slots = gh ? this.$slots.default() : this.$slots.default;
        }
        return h('div', slots);
    }
    custom() {
        this.updated();
    }
    export(type, fileName) {
        return this.ej2Instances.export(type, fileName);
    }
    print(id) {
        return this.ej2Instances.print(id);
    }
    setAnnotationValue(annotationIndex, content) {
        return this.ej2Instances.setAnnotationValue(annotationIndex, content);
    }
    titleTooltip(event, x, y, isTouch) {
        return this.ej2Instances.titleTooltip(event, x, y, isTouch);
    }
};
AccumulationChartComponent = __decorate$16([
    EJComponentDecorator({
        props: properties$1,
        model: {
            event: 'modelchanged'
        }
    }, isExecute$16)
    /* Start Options({
        props: props,
        watch: watch,
        emits: emitProbs,
        provide: function provide() {
            return {
                custom: this.custom
            };
        }
    }) End */
], AccumulationChartComponent);
const AccumulationChartPlugin = {
    name: 'ejs-accumulationchart',
    install(Vue$$1) {
        Vue$$1.component(AccumulationChartPlugin.name, AccumulationChartComponent);
        Vue$$1.component(AccumulationSeriesPlugin.name, AccumulationSeriesDirective);
        Vue$$1.component(AccumulationSeriesCollectionPlugin.name, AccumulationSeriesCollectionDirective);
        Vue$$1.component(AccumulationAnnotationPlugin.name, AccumulationAnnotationDirective);
        Vue$$1.component(AccumulationAnnotationsPlugin.name, AccumulationAnnotationsDirective);
    }
};

var __decorate$17 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$17 = gh ? false : true;
let vueImport$15;
if (!isExecute$17 || parseInt(allVue.version) < 3) {
    vueImport$15 = Vue;
}
else {
    vueImport$15 = Vue$1;
}
let RangenavigatorSeriesCollectionDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class RangenavigatorSeriesCollectionDirective extends vueImport$15 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-rangenavigator-series-collection';
    }
};
RangenavigatorSeriesCollectionDirective = __decorate$17([
    EJComponentDecorator({}, isExecute$17)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], RangenavigatorSeriesCollectionDirective);
const RangenavigatorSeriesCollectionPlugin = {
    name: 'e-rangenavigator-series-collection',
    install(Vue$$1) {
        Vue$$1.component(RangenavigatorSeriesCollectionPlugin.name, RangenavigatorSeriesCollectionDirective);
    }
};
let RangenavigatorSeriesDirective = class RangenavigatorSeriesDirective extends vueImport$15 {
    render() {
        return;
    }
    getTag() {
        return 'e-rangenavigator-series';
    }
};
RangenavigatorSeriesDirective = __decorate$17([
    EJComponentDecorator({}, isExecute$17)
], RangenavigatorSeriesDirective);
const RangenavigatorSeriesPlugin = {
    name: 'e-rangenavigator-series',
    install(Vue$$1) {
        Vue$$1.component(RangenavigatorSeriesPlugin.name, RangenavigatorSeriesDirective);
    }
};

var __decorate$18 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const properties$2 = ['isLazyUpdate', 'allowIntervalData', 'allowSnapping', 'animationDuration', 'background', 'dataSource', 'disableRangeSelector', 'enableDeferredUpdate', 'enableGrouping', 'enablePersistence', 'enableRtl', 'groupBy', 'height', 'interval', 'intervalType', 'labelFormat', 'labelIntersectAction', 'labelPosition', 'labelStyle', 'locale', 'logBase', 'majorGridLines', 'majorTickLines', 'margin', 'maximum', 'minimum', 'navigatorBorder', 'navigatorStyleSettings', 'periodSelectorSettings', 'query', 'secondaryLabelAlignment', 'series', 'skeleton', 'skeletonType', 'theme', 'tickPosition', 'tooltip', 'useGroupingSeparator', 'value', 'valueType', 'width', 'xName', 'yName', 'beforePrint', 'beforeResize', 'changed', 'labelRender', 'load', 'loaded', 'resized', 'selectorRender', 'tooltipRender'];
const modelProps$2 = ['dataSource'];
const testProp$2 = getProps({ props: properties$2 });
const props$2 = testProp$2[0];
const watch$2 = testProp$2[1];
const emitProbs$2 = Object.keys(watch$2);
emitProbs$2.push('modelchanged');
for (let props of modelProps$2) {
    emitProbs$2.push('update:' + props);
}
const isExecute$18 = gh ? false : true;
/**
 * Represents Vuejs RangeNavigator Component
 * ```vue
 * <ejs-rangenavigator></ejs-rangenavigator>
 * ```
 */
let RangeNavigatorComponent = 
/* Start Options({
    props: props,
    watch: watch,
    emits: emitProbs,
    provide: function provide() {
        return {
            custom: this.custom
        };
    }
}) End */
class RangeNavigatorComponent extends ComponentBase {
    constructor() {
        super(arguments);
        this.propKeys = properties$2;
        this.models = modelProps$2;
        this.hasChildDirective = true;
        this.hasInjectedModules = true;
        this.tagMapper = { "e-rangenavigator-series-collection": "e-rangenavigator-series" };
        this.tagNameMapper = { "e-rangenavigator-series-collection": "e-series" };
        this.isVue3 = !isExecute$18;
        this.ej2Instances = new RangeNavigator({});
        this.ej2Instances._trigger = this.ej2Instances.trigger;
        this.ej2Instances.trigger = this.trigger;
        this.bindProperties();
        this.ej2Instances._setProperties = this.ej2Instances.setProperties;
        this.ej2Instances.setProperties = this.setProperties;
        this.ej2Instances.clearTemplate = this.clearTemplate;
        this.updated = this.updated;
    }
    clearTemplate(templateNames) {
        if (!templateNames) {
            templateNames = Object.keys(this.templateCollection || {});
        }
        if (templateNames.length && this.templateCollection) {
            for (let tempName of templateNames) {
                let elementCollection = this.templateCollection[tempName];
                if (elementCollection && elementCollection.length) {
                    for (let ele of elementCollection) {
                        let destroy = getValue('__vue__.$destroy', ele);
                        if (destroy) {
                            ele.__vue__.$destroy();
                        }
                        if (ele.innerHTML) {
                            ele.innerHTML = '';
                        }
                    }
                    delete this.templateCollection[tempName];
                }
            }
        }
    }
    setProperties(prop, muteOnChange) {
        if (this.isVue3) {
            this.models = !this.models ? this.ej2Instances.referModels : this.models;
        }
        if (this.ej2Instances && this.ej2Instances._setProperties) {
            this.ej2Instances._setProperties(prop, muteOnChange);
        }
        if (prop && this.models && this.models.length) {
            Object.keys(prop).map((key) => {
                this.models.map((model) => {
                    if ((key === model) && !(/datasource/i.test(key))) {
                        if (this.isVue3) {
                            this.ej2Instances.vueInstance.$emit('update:' + key, prop[key]);
                        }
                        else {
                            this.$emit('update:' + key, prop[key]);
                            this.$emit('modelchanged', prop[key]);
                        }
                    }
                });
            });
        }
    }
    trigger(eventName, eventProp, successHandler) {
        if (!isExecute$18) {
            this.models = !this.models ? this.ej2Instances.referModels : this.models;
        }
        if ((eventName === 'change' || eventName === 'input') && this.models && (this.models.length !== 0)) {
            let key = this.models.toString().match(/checked|value/) || [];
            let propKey = key[0];
            if (eventProp && key && !isUndefined(eventProp[propKey])) {
                if (!isExecute$18) {
                    this.ej2Instances.vueInstance.$emit('update:' + propKey, eventProp[propKey]);
                    this.ej2Instances.vueInstance.$emit('modelchanged', eventProp[propKey]);
                }
                else {
                    if (eventName === 'change' || (this.$props && !this.$props.isLazyUpdate)) {
                        this.$emit('update:' + propKey, eventProp[propKey]);
                        this.$emit('modelchanged', eventProp[propKey]);
                    }
                }
            }
        }
        else if ((eventName === 'actionBegin' && eventProp.requestType === 'dateNavigate') && this.models && (this.models.length !== 0)) {
            let key = this.models.toString().match(/currentView|selectedDate/) || [];
            let propKey = key[0];
            if (eventProp && key && !isUndefined(eventProp[propKey])) {
                if (!isExecute$18) {
                    this.ej2Instances.vueInstance.$emit('update:' + propKey, eventProp[propKey]);
                    this.ej2Instances.vueInstance.$emit('modelchanged', eventProp[propKey]);
                }
                else {
                    this.$emit('update:' + propKey, eventProp[propKey]);
                    this.$emit('modelchanged', eventProp[propKey]);
                }
            }
        }
        if ((this.ej2Instances && this.ej2Instances._trigger)) {
            this.ej2Instances._trigger(eventName, eventProp, successHandler);
        }
    }
    render(createElement) {
        let h = gh || createElement;
        let slots = null;
        if (!isNullOrUndefined(this.$slots.default)) {
            slots = gh ? this.$slots.default() : this.$slots.default;
        }
        return h('div', slots);
    }
    custom() {
        this.updated();
    }
    createSecondaryElement() {
        return this.ej2Instances.createSecondaryElement();
    }
    export(type, fileName, orientation, controls, width, height, isVertical) {
        return this.ej2Instances.export(type, fileName, orientation, controls, width, height, isVertical);
    }
    print(id) {
        return this.ej2Instances.print(id);
    }
    renderChart(resize) {
        return this.ej2Instances.renderChart(resize);
    }
};
RangeNavigatorComponent = __decorate$18([
    EJComponentDecorator({
        props: properties$2,
        model: {
            event: 'modelchanged'
        }
    }, isExecute$18)
    /* Start Options({
        props: props,
        watch: watch,
        emits: emitProbs,
        provide: function provide() {
            return {
                custom: this.custom
            };
        }
    }) End */
], RangeNavigatorComponent);
const RangeNavigatorPlugin = {
    name: 'ejs-rangenavigator',
    install(Vue$$1) {
        Vue$$1.component(RangeNavigatorPlugin.name, RangeNavigatorComponent);
        Vue$$1.component(RangenavigatorSeriesPlugin.name, RangenavigatorSeriesDirective);
        Vue$$1.component(RangenavigatorSeriesCollectionPlugin.name, RangenavigatorSeriesCollectionDirective);
    }
};

var __decorate$19 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$19 = gh ? false : true;
let vueImport$16;
if (!isExecute$19 || parseInt(allVue.version) < 3) {
    vueImport$16 = Vue;
}
else {
    vueImport$16 = Vue$1;
}
let RangeBandSettingsDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class RangeBandSettingsDirective extends vueImport$16 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-rangeBandSettings';
    }
};
RangeBandSettingsDirective = __decorate$19([
    EJComponentDecorator({}, isExecute$19)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], RangeBandSettingsDirective);
const RangeBandSettingsPlugin = {
    name: 'e-rangeBandSettings',
    install(Vue$$1) {
        Vue$$1.component(RangeBandSettingsPlugin.name, RangeBandSettingsDirective);
    }
};
let RangeBandSettingDirective = class RangeBandSettingDirective extends vueImport$16 {
    render() {
        return;
    }
    getTag() {
        return 'e-rangeBandSetting';
    }
};
RangeBandSettingDirective = __decorate$19([
    EJComponentDecorator({}, isExecute$19)
], RangeBandSettingDirective);
const RangeBandSettingPlugin = {
    name: 'e-rangeBandSetting',
    install(Vue$$1) {
        Vue$$1.component(RangeBandSettingPlugin.name, RangeBandSettingDirective);
    }
};

var __decorate$20 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const properties$3 = ['isLazyUpdate', 'axisSettings', 'border', 'containerArea', 'dataLabelSettings', 'dataSource', 'enablePersistence', 'enableRtl', 'endPointColor', 'fill', 'format', 'height', 'highPointColor', 'lineWidth', 'locale', 'lowPointColor', 'markerSettings', 'negativePointColor', 'opacity', 'padding', 'palette', 'query', 'rangeBandSettings', 'rangePadding', 'startPointColor', 'theme', 'tiePointColor', 'tooltipSettings', 'type', 'useGroupingSeparator', 'valueType', 'width', 'xName', 'yName', 'axisRendering', 'dataLabelRendering', 'load', 'loaded', 'markerRendering', 'pointRegionMouseClick', 'pointRegionMouseMove', 'pointRendering', 'resize', 'seriesRendering', 'sparklineMouseClick', 'sparklineMouseMove', 'tooltipInitialize'];
const modelProps$3 = [];
const testProp$3 = getProps({ props: properties$3 });
const props$3 = testProp$3[0];
const watch$3 = testProp$3[1];
const emitProbs$3 = Object.keys(watch$3);
emitProbs$3.push('modelchanged');
for (let props of modelProps$3) {
    emitProbs$3.push('update:' + props);
}
const isExecute$20 = gh ? false : true;
/**
 * Represents Vuejs Sparkline Component
 * ```vue
 * <ejs-sparkline></ejs-sparkline>
 * ```
 */
let SparklineComponent = 
/* Start Options({
    props: props,
    watch: watch,
    emits: emitProbs,
    provide: function provide() {
        return {
            custom: this.custom
        };
    }
}) End */
class SparklineComponent extends ComponentBase {
    constructor() {
        super(arguments);
        this.propKeys = properties$3;
        this.models = modelProps$3;
        this.hasChildDirective = true;
        this.hasInjectedModules = true;
        this.tagMapper = { "e-rangeBandSettings": "e-rangeBandSetting" };
        this.tagNameMapper = {};
        this.isVue3 = !isExecute$20;
        this.ej2Instances = new Sparkline({});
        this.bindProperties();
        this.ej2Instances._setProperties = this.ej2Instances.setProperties;
        this.ej2Instances.setProperties = this.setProperties;
        this.ej2Instances.clearTemplate = this.clearTemplate;
        this.updated = this.updated;
    }
    clearTemplate(templateNames) {
        if (!templateNames) {
            templateNames = Object.keys(this.templateCollection || {});
        }
        if (templateNames.length && this.templateCollection) {
            for (let tempName of templateNames) {
                let elementCollection = this.templateCollection[tempName];
                if (elementCollection && elementCollection.length) {
                    for (let ele of elementCollection) {
                        let destroy = getValue('__vue__.$destroy', ele);
                        if (destroy) {
                            ele.__vue__.$destroy();
                        }
                        if (ele.innerHTML) {
                            ele.innerHTML = '';
                        }
                    }
                    delete this.templateCollection[tempName];
                }
            }
        }
    }
    setProperties(prop, muteOnChange) {
        if (this.isVue3) {
            this.models = !this.models ? this.ej2Instances.referModels : this.models;
        }
        if (this.ej2Instances && this.ej2Instances._setProperties) {
            this.ej2Instances._setProperties(prop, muteOnChange);
        }
        if (prop && this.models && this.models.length) {
            Object.keys(prop).map((key) => {
                this.models.map((model) => {
                    if ((key === model) && !(/datasource/i.test(key))) {
                        if (this.isVue3) {
                            this.ej2Instances.vueInstance.$emit('update:' + key, prop[key]);
                        }
                        else {
                            this.$emit('update:' + key, prop[key]);
                            this.$emit('modelchanged', prop[key]);
                        }
                    }
                });
            });
        }
    }
    render(createElement) {
        let h = gh || createElement;
        let slots = null;
        if (!isNullOrUndefined(this.$slots.default)) {
            slots = gh ? this.$slots.default() : this.$slots.default;
        }
        return h('div', slots);
    }
    custom() {
        this.updated();
    }
    renderSparkline() {
        return this.ej2Instances.renderSparkline();
    }
};
SparklineComponent = __decorate$20([
    EJComponentDecorator({
        props: properties$3
    }, isExecute$20)
    /* Start Options({
        props: props,
        watch: watch,
        emits: emitProbs,
        provide: function provide() {
            return {
                custom: this.custom
            };
        }
    }) End */
], SparklineComponent);
const SparklinePlugin = {
    name: 'ejs-sparkline',
    install(Vue$$1) {
        Vue$$1.component(SparklinePlugin.name, SparklineComponent);
        Vue$$1.component(RangeBandSettingPlugin.name, RangeBandSettingDirective);
        Vue$$1.component(RangeBandSettingsPlugin.name, RangeBandSettingsDirective);
    }
};

var __decorate$21 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$21 = gh ? false : true;
let vueImport$17;
if (!isExecute$21 || parseInt(allVue.version) < 3) {
    vueImport$17 = Vue;
}
else {
    vueImport$17 = Vue$1;
}
let SmithchartSeriesCollectionDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class SmithchartSeriesCollectionDirective extends vueImport$17 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-seriesCollection';
    }
};
SmithchartSeriesCollectionDirective = __decorate$21([
    EJComponentDecorator({}, isExecute$21)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], SmithchartSeriesCollectionDirective);
const SmithchartSeriesCollectionPlugin = {
    name: 'e-seriesCollection',
    install(Vue$$1) {
        Vue$$1.component(SmithchartSeriesCollectionPlugin.name, SmithchartSeriesCollectionDirective);
    }
};
let SmithchartSeriesDirective = class SmithchartSeriesDirective extends vueImport$17 {
    render() {
        return;
    }
    getTag() {
        return 'e-series';
    }
};
SmithchartSeriesDirective = __decorate$21([
    EJComponentDecorator({}, isExecute$21)
], SmithchartSeriesDirective);
const SmithchartSeriesPlugin = {
    name: 'e-series',
    install(Vue$$1) {
        Vue$$1.component(SmithchartSeriesPlugin.name, SmithchartSeriesDirective);
    }
};

var __decorate$22 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const properties$4 = ['isLazyUpdate', 'background', 'border', 'elementSpacing', 'enablePersistence', 'enableRtl', 'font', 'height', 'horizontalAxis', 'legendSettings', 'locale', 'margin', 'radialAxis', 'radius', 'renderType', 'series', 'theme', 'title', 'width', 'animationComplete', 'axisLabelRender', 'beforePrint', 'legendRender', 'load', 'loaded', 'seriesRender', 'subtitleRender', 'textRender', 'titleRender', 'tooltipRender'];
const modelProps$4 = [];
const testProp$4 = getProps({ props: properties$4 });
const props$4 = testProp$4[0];
const watch$4 = testProp$4[1];
const emitProbs$4 = Object.keys(watch$4);
emitProbs$4.push('modelchanged');
for (let props of modelProps$4) {
    emitProbs$4.push('update:' + props);
}
const isExecute$22 = gh ? false : true;
/**
 * Represents Vuejs Smithchart Component
 * ```vue
 * <ejs-smithchart></ejs-smithchart>
 * ```
 */
let SmithchartComponent = 
/* Start Options({
    props: props,
    watch: watch,
    emits: emitProbs,
    provide: function provide() {
        return {
            custom: this.custom
        };
    }
}) End */
class SmithchartComponent extends ComponentBase {
    constructor() {
        super(arguments);
        this.propKeys = properties$4;
        this.models = modelProps$4;
        this.hasChildDirective = true;
        this.hasInjectedModules = true;
        this.tagMapper = { "e-seriesCollection": "e-series" };
        this.tagNameMapper = { "e-seriesCollection": "e-series" };
        this.isVue3 = !isExecute$22;
        this.ej2Instances = new Smithchart({});
        this.bindProperties();
        this.ej2Instances._setProperties = this.ej2Instances.setProperties;
        this.ej2Instances.setProperties = this.setProperties;
        this.ej2Instances.clearTemplate = this.clearTemplate;
        this.updated = this.updated;
    }
    clearTemplate(templateNames) {
        if (!templateNames) {
            templateNames = Object.keys(this.templateCollection || {});
        }
        if (templateNames.length && this.templateCollection) {
            for (let tempName of templateNames) {
                let elementCollection = this.templateCollection[tempName];
                if (elementCollection && elementCollection.length) {
                    for (let ele of elementCollection) {
                        let destroy = getValue('__vue__.$destroy', ele);
                        if (destroy) {
                            ele.__vue__.$destroy();
                        }
                        if (ele.innerHTML) {
                            ele.innerHTML = '';
                        }
                    }
                    delete this.templateCollection[tempName];
                }
            }
        }
    }
    setProperties(prop, muteOnChange) {
        if (this.isVue3) {
            this.models = !this.models ? this.ej2Instances.referModels : this.models;
        }
        if (this.ej2Instances && this.ej2Instances._setProperties) {
            this.ej2Instances._setProperties(prop, muteOnChange);
        }
        if (prop && this.models && this.models.length) {
            Object.keys(prop).map((key) => {
                this.models.map((model) => {
                    if ((key === model) && !(/datasource/i.test(key))) {
                        if (this.isVue3) {
                            this.ej2Instances.vueInstance.$emit('update:' + key, prop[key]);
                        }
                        else {
                            this.$emit('update:' + key, prop[key]);
                            this.$emit('modelchanged', prop[key]);
                        }
                    }
                });
            });
        }
    }
    render(createElement) {
        let h = gh || createElement;
        let slots = null;
        if (!isNullOrUndefined(this.$slots.default)) {
            slots = gh ? this.$slots.default() : this.$slots.default;
        }
        return h('div', slots);
    }
    custom() {
        this.updated();
    }
    export(type, fileName, orientation) {
        return this.ej2Instances.export(type, fileName, orientation);
    }
    mouseEnd(e) {
        return this.ej2Instances.mouseEnd(e);
    }
    mouseMove(e) {
        return this.ej2Instances.mouseMove(e);
    }
    print(id) {
        return this.ej2Instances.print(id);
    }
    smithchartOnClick(e) {
        return this.ej2Instances.smithchartOnClick(e);
    }
    smithchartOnResize() {
        return this.ej2Instances.smithchartOnResize();
    }
};
SmithchartComponent = __decorate$22([
    EJComponentDecorator({
        props: properties$4
    }, isExecute$22)
    /* Start Options({
        props: props,
        watch: watch,
        emits: emitProbs,
        provide: function provide() {
            return {
                custom: this.custom
            };
        }
    }) End */
], SmithchartComponent);
const SmithchartPlugin = {
    name: 'ejs-smithchart',
    install(Vue$$1) {
        Vue$$1.component(SmithchartPlugin.name, SmithchartComponent);
        Vue$$1.component(SmithchartSeriesPlugin.name, SmithchartSeriesDirective);
        Vue$$1.component(SmithchartSeriesCollectionPlugin.name, SmithchartSeriesCollectionDirective);
    }
};

var __decorate$23 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$23 = gh ? false : true;
let vueImport$18;
if (!isExecute$23 || parseInt(allVue.version) < 3) {
    vueImport$18 = Vue;
}
else {
    vueImport$18 = Vue$1;
}
let StockChartTrendlinesDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class StockChartTrendlinesDirective extends vueImport$18 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-trendlines';
    }
};
StockChartTrendlinesDirective = __decorate$23([
    EJComponentDecorator({}, isExecute$23)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], StockChartTrendlinesDirective);
const StockChartTrendlinesPlugin = {
    name: 'e-trendlines',
    install(Vue$$1) {
        Vue$$1.component(StockChartTrendlinesPlugin.name, StockChartTrendlinesDirective);
    }
};
let StockChartTrendlineDirective = class StockChartTrendlineDirective extends vueImport$18 {
    render() {
        return;
    }
    getTag() {
        return 'e-trendline';
    }
};
StockChartTrendlineDirective = __decorate$23([
    EJComponentDecorator({}, isExecute$23)
], StockChartTrendlineDirective);
const StockChartTrendlinePlugin = {
    name: 'e-trendline',
    install(Vue$$1) {
        Vue$$1.component(StockChartTrendlinePlugin.name, StockChartTrendlineDirective);
    }
};

var __decorate$24 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$24 = gh ? false : true;
let vueImport$19;
if (!isExecute$24 || parseInt(allVue.version) < 3) {
    vueImport$19 = Vue;
}
else {
    vueImport$19 = Vue$1;
}
let StockChartSeriesCollectionDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class StockChartSeriesCollectionDirective extends vueImport$19 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-stockchart-series-collection';
    }
};
StockChartSeriesCollectionDirective = __decorate$24([
    EJComponentDecorator({}, isExecute$24)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], StockChartSeriesCollectionDirective);
const StockChartSeriesCollectionPlugin = {
    name: 'e-stockchart-series-collection',
    install(Vue$$1) {
        Vue$$1.component(StockChartSeriesCollectionPlugin.name, StockChartSeriesCollectionDirective);
    }
};
let StockChartSeriesDirective = class StockChartSeriesDirective extends vueImport$19 {
    render() {
        return;
    }
    getTag() {
        return 'e-stockchart-series';
    }
};
StockChartSeriesDirective = __decorate$24([
    EJComponentDecorator({}, isExecute$24)
], StockChartSeriesDirective);
const StockChartSeriesPlugin = {
    name: 'e-stockchart-series',
    install(Vue$$1) {
        Vue$$1.component(StockChartSeriesPlugin.name, StockChartSeriesDirective);
    }
};

var __decorate$25 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$25 = gh ? false : true;
let vueImport$20;
if (!isExecute$25 || parseInt(allVue.version) < 3) {
    vueImport$20 = Vue;
}
else {
    vueImport$20 = Vue$1;
}
let StockChartAxesDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class StockChartAxesDirective extends vueImport$20 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-stockchart-axes';
    }
};
StockChartAxesDirective = __decorate$25([
    EJComponentDecorator({}, isExecute$25)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], StockChartAxesDirective);
const StockChartAxesPlugin = {
    name: 'e-stockchart-axes',
    install(Vue$$1) {
        Vue$$1.component(StockChartAxesPlugin.name, StockChartAxesDirective);
    }
};
let StockChartAxisDirective = class StockChartAxisDirective extends vueImport$20 {
    render() {
        return;
    }
    getTag() {
        return 'e-stockchart-axis';
    }
};
StockChartAxisDirective = __decorate$25([
    EJComponentDecorator({}, isExecute$25)
], StockChartAxisDirective);
const StockChartAxisPlugin = {
    name: 'e-stockchart-axis',
    install(Vue$$1) {
        Vue$$1.component(StockChartAxisPlugin.name, StockChartAxisDirective);
    }
};

var __decorate$26 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$26 = gh ? false : true;
let vueImport$21;
if (!isExecute$26 || parseInt(allVue.version) < 3) {
    vueImport$21 = Vue;
}
else {
    vueImport$21 = Vue$1;
}
let StockChartRowsDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class StockChartRowsDirective extends vueImport$21 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-stockchart-rows';
    }
};
StockChartRowsDirective = __decorate$26([
    EJComponentDecorator({}, isExecute$26)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], StockChartRowsDirective);
const StockChartRowsPlugin = {
    name: 'e-stockchart-rows',
    install(Vue$$1) {
        Vue$$1.component(StockChartRowsPlugin.name, StockChartRowsDirective);
    }
};
let StockChartRowDirective = class StockChartRowDirective extends vueImport$21 {
    render() {
        return;
    }
    getTag() {
        return 'e-stockchart-row';
    }
};
StockChartRowDirective = __decorate$26([
    EJComponentDecorator({}, isExecute$26)
], StockChartRowDirective);
const StockChartRowPlugin = {
    name: 'e-stockchart-row',
    install(Vue$$1) {
        Vue$$1.component(StockChartRowPlugin.name, StockChartRowDirective);
    }
};

var __decorate$27 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$27 = gh ? false : true;
let vueImport$22;
if (!isExecute$27 || parseInt(allVue.version) < 3) {
    vueImport$22 = Vue;
}
else {
    vueImport$22 = Vue$1;
}
let StockChartAnnotationsDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class StockChartAnnotationsDirective extends vueImport$22 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-stockchart-annotations';
    }
};
StockChartAnnotationsDirective = __decorate$27([
    EJComponentDecorator({}, isExecute$27)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], StockChartAnnotationsDirective);
const StockChartAnnotationsPlugin = {
    name: 'e-stockchart-annotations',
    install(Vue$$1) {
        Vue$$1.component(StockChartAnnotationsPlugin.name, StockChartAnnotationsDirective);
    }
};
/**
 * `e-annotation` directive represent a annotation of the VueJS Chart.
 * It must be contained in a Chart component(`ejs-chart`).
 * ```vue
 * <ejs-stockchart>
 *   <e-stockchart-annotations>
 *    <e-annotation content='ID' />
 *    <e-annotation content='ID' />
 *   </e-annotations>
 * </ejs-chart>
 * ```
 */
let StockChartAnnotationDirective = class StockChartAnnotationDirective extends vueImport$22 {
    render() {
        return;
    }
    getTag() {
        return 'e-stockchart-annotation';
    }
};
StockChartAnnotationDirective = __decorate$27([
    EJComponentDecorator({}, isExecute$27)
], StockChartAnnotationDirective);
const StockChartAnnotationPlugin = {
    name: 'e-stockchart-annotation',
    install(Vue$$1) {
        Vue$$1.component(StockChartAnnotationPlugin.name, StockChartAnnotationDirective);
    }
};

var __decorate$28 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$28 = gh ? false : true;
let vueImport$23;
if (!isExecute$28 || parseInt(allVue.version) < 3) {
    vueImport$23 = Vue;
}
else {
    vueImport$23 = Vue$1;
}
let StockChartSelectedDataIndexesDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class StockChartSelectedDataIndexesDirective extends vueImport$23 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-stockchart-selectedDataIndexes';
    }
};
StockChartSelectedDataIndexesDirective = __decorate$28([
    EJComponentDecorator({}, isExecute$28)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], StockChartSelectedDataIndexesDirective);
const StockChartSelectedDataIndexesPlugin = {
    name: 'e-stockchart-selectedDataIndexes',
    install(Vue$$1) {
        Vue$$1.component(StockChartSelectedDataIndexesPlugin.name, StockChartSelectedDataIndexesDirective);
    }
};
let StockChartSelectedDataIndexDirective = class StockChartSelectedDataIndexDirective extends vueImport$23 {
    render() {
        return;
    }
    getTag() {
        return 'e-stockchart-selectedDataIndex';
    }
};
StockChartSelectedDataIndexDirective = __decorate$28([
    EJComponentDecorator({}, isExecute$28)
], StockChartSelectedDataIndexDirective);
const StockChartSelectedDataIndexPlugin = {
    name: 'e-stockchart-selectedDataIndex',
    install(Vue$$1) {
        Vue$$1.component(StockChartSelectedDataIndexPlugin.name, StockChartSelectedDataIndexDirective);
    }
};

var __decorate$29 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$29 = gh ? false : true;
let vueImport$24;
if (!isExecute$29 || parseInt(allVue.version) < 3) {
    vueImport$24 = Vue;
}
else {
    vueImport$24 = Vue$1;
}
let StockChartPeriodsDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class StockChartPeriodsDirective extends vueImport$24 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-stockchart-periods';
    }
};
StockChartPeriodsDirective = __decorate$29([
    EJComponentDecorator({}, isExecute$29)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], StockChartPeriodsDirective);
const StockChartPeriodsPlugin = {
    name: 'e-stockchart-periods',
    install(Vue$$1) {
        Vue$$1.component(StockChartPeriodsPlugin.name, StockChartPeriodsDirective);
    }
};
let StockChartPeriodDirective = class StockChartPeriodDirective extends vueImport$24 {
    render() {
        return;
    }
    getTag() {
        return 'e-stockchart-period';
    }
};
StockChartPeriodDirective = __decorate$29([
    EJComponentDecorator({}, isExecute$29)
], StockChartPeriodDirective);
const StockChartPeriodPlugin = {
    name: 'e-stockchart-period',
    install(Vue$$1) {
        Vue$$1.component(StockChartPeriodPlugin.name, StockChartPeriodDirective);
    }
};

var __decorate$30 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$30 = gh ? false : true;
let vueImport$25;
if (!isExecute$30 || parseInt(allVue.version) < 3) {
    vueImport$25 = Vue;
}
else {
    vueImport$25 = Vue$1;
}
let StockEventsDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class StockEventsDirective extends vueImport$25 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-stockchart-stockevents';
    }
};
StockEventsDirective = __decorate$30([
    EJComponentDecorator({}, isExecute$30)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], StockEventsDirective);
const StockEventsPlugin = {
    name: 'e-stockchart-stockevents',
    install(Vue$$1) {
        Vue$$1.component(StockEventsPlugin.name, StockEventsDirective);
    }
};
let StockEventDirective = class StockEventDirective extends vueImport$25 {
    render() {
        return;
    }
    getTag() {
        return 'e-stockchart-stockevent';
    }
};
StockEventDirective = __decorate$30([
    EJComponentDecorator({}, isExecute$30)
], StockEventDirective);
const StockEventPlugin = {
    name: 'e-stockchart-stockevent',
    install(Vue$$1) {
        Vue$$1.component(StockEventPlugin.name, StockEventDirective);
    }
};

var __decorate$31 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$31 = gh ? false : true;
let vueImport$26;
if (!isExecute$31 || parseInt(allVue.version) < 3) {
    vueImport$26 = Vue;
}
else {
    vueImport$26 = Vue$1;
}
let StockChartIndicatorsDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class StockChartIndicatorsDirective extends vueImport$26 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-stockchart-indicators';
    }
};
StockChartIndicatorsDirective = __decorate$31([
    EJComponentDecorator({}, isExecute$31)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], StockChartIndicatorsDirective);
const StockChartIndicatorsPlugin = {
    name: 'e-stockchart-indicators',
    install(Vue$$1) {
        Vue$$1.component(StockChartIndicatorsPlugin.name, StockChartIndicatorsDirective);
    }
};
let StockChartIndicatorDirective = class StockChartIndicatorDirective extends vueImport$26 {
    render() {
        return;
    }
    getTag() {
        return 'e-stockchart-indicator';
    }
};
StockChartIndicatorDirective = __decorate$31([
    EJComponentDecorator({}, isExecute$31)
], StockChartIndicatorDirective);
const StockChartIndicatorPlugin = {
    name: 'e-stockchart-indicator',
    install(Vue$$1) {
        Vue$$1.component(StockChartIndicatorPlugin.name, StockChartIndicatorDirective);
    }
};

var __decorate$32 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const properties$5 = ['isLazyUpdate', 'annotations', 'axes', 'background', 'border', 'chartArea', 'crosshair', 'dataSource', 'enableCustomRange', 'enablePeriodSelector', 'enablePersistence', 'enableRtl', 'enableSelector', 'exportType', 'height', 'indicatorType', 'indicators', 'isMultiSelect', 'isSelect', 'isTransposed', 'legendSettings', 'locale', 'margin', 'periods', 'primaryXAxis', 'primaryYAxis', 'rows', 'selectedDataIndexes', 'selectionMode', 'series', 'seriesType', 'stockEvents', 'theme', 'title', 'titleStyle', 'tooltip', 'trendlineType', 'width', 'zoomSettings', 'axisLabelRender', 'legendClick', 'legendRender', 'load', 'loaded', 'onZooming', 'pointClick', 'pointMove', 'rangeChange', 'selectorRender', 'seriesRender', 'stockChartMouseClick', 'stockChartMouseDown', 'stockChartMouseLeave', 'stockChartMouseMove', 'stockChartMouseUp', 'stockEventRender', 'tooltipRender'];
const modelProps$5 = ['dataSource'];
const testProp$5 = getProps({ props: properties$5 });
const props$5 = testProp$5[0];
const watch$5 = testProp$5[1];
const emitProbs$5 = Object.keys(watch$5);
emitProbs$5.push('modelchanged');
for (let props of modelProps$5) {
    emitProbs$5.push('update:' + props);
}
const isExecute$32 = gh ? false : true;
/**
 * Represents Vuejs chart Component
 * ```vue
 * <ejs-stockchart></ejs-stockchart>
 * ```
 */
let StockChartComponent = 
/* Start Options({
    props: props,
    watch: watch,
    emits: emitProbs,
    provide: function provide() {
        return {
            custom: this.custom
        };
    }
}) End */
class StockChartComponent extends ComponentBase {
    constructor() {
        super(arguments);
        this.propKeys = properties$5;
        this.models = modelProps$5;
        this.hasChildDirective = true;
        this.hasInjectedModules = true;
        this.tagMapper = { "e-stockchart-series-collection": { "e-stockchart-series": { "e-trendlines": "e-trendline" } }, "e-stockchart-axes": "e-stockchart-axis", "e-stockchart-rows": "e-stockchart-row", "e-stockchart-annotations": "e-stockchart-annotation", "e-stockchart-selectedDataIndexes": "e-stockchart-selectedDataIndex", "e-stockchart-periods": "e-stockchart-period", "e-stockchart-stockevents": "e-stockchart-stockevent", "e-stockchart-indicators": "e-stockchart-indicator" };
        this.tagNameMapper = { "e-stockchart-series-collection": "e-series", "e-stockchart-axes": "e-axes", "e-stockchart-rows": "e-rows", "e-stockchart-annotations": "e-annotations", "e-stockchart-selectedDataIndexes": "e-selectedDataIndexes", "e-stockchart-periods": "e-periods", "e-stockchart-stockevents": "e-stockEvents", "e-stockchart-indicators": "e-indicators" };
        this.isVue3 = !isExecute$32;
        this.ej2Instances = new StockChart({});
        this.ej2Instances._trigger = this.ej2Instances.trigger;
        this.ej2Instances.trigger = this.trigger;
        this.bindProperties();
        this.ej2Instances._setProperties = this.ej2Instances.setProperties;
        this.ej2Instances.setProperties = this.setProperties;
        this.ej2Instances.clearTemplate = this.clearTemplate;
        this.updated = this.updated;
    }
    clearTemplate(templateNames) {
        if (!templateNames) {
            templateNames = Object.keys(this.templateCollection || {});
        }
        if (templateNames.length && this.templateCollection) {
            for (let tempName of templateNames) {
                let elementCollection = this.templateCollection[tempName];
                if (elementCollection && elementCollection.length) {
                    for (let ele of elementCollection) {
                        let destroy = getValue('__vue__.$destroy', ele);
                        if (destroy) {
                            ele.__vue__.$destroy();
                        }
                        if (ele.innerHTML) {
                            ele.innerHTML = '';
                        }
                    }
                    delete this.templateCollection[tempName];
                }
            }
        }
    }
    setProperties(prop, muteOnChange) {
        if (this.isVue3) {
            this.models = !this.models ? this.ej2Instances.referModels : this.models;
        }
        if (this.ej2Instances && this.ej2Instances._setProperties) {
            this.ej2Instances._setProperties(prop, muteOnChange);
        }
        if (prop && this.models && this.models.length) {
            Object.keys(prop).map((key) => {
                this.models.map((model) => {
                    if ((key === model) && !(/datasource/i.test(key))) {
                        if (this.isVue3) {
                            this.ej2Instances.vueInstance.$emit('update:' + key, prop[key]);
                        }
                        else {
                            this.$emit('update:' + key, prop[key]);
                            this.$emit('modelchanged', prop[key]);
                        }
                    }
                });
            });
        }
    }
    trigger(eventName, eventProp, successHandler) {
        if (!isExecute$32) {
            this.models = !this.models ? this.ej2Instances.referModels : this.models;
        }
        if ((eventName === 'change' || eventName === 'input') && this.models && (this.models.length !== 0)) {
            let key = this.models.toString().match(/checked|value/) || [];
            let propKey = key[0];
            if (eventProp && key && !isUndefined(eventProp[propKey])) {
                if (!isExecute$32) {
                    this.ej2Instances.vueInstance.$emit('update:' + propKey, eventProp[propKey]);
                    this.ej2Instances.vueInstance.$emit('modelchanged', eventProp[propKey]);
                }
                else {
                    if (eventName === 'change' || (this.$props && !this.$props.isLazyUpdate)) {
                        this.$emit('update:' + propKey, eventProp[propKey]);
                        this.$emit('modelchanged', eventProp[propKey]);
                    }
                }
            }
        }
        else if ((eventName === 'actionBegin' && eventProp.requestType === 'dateNavigate') && this.models && (this.models.length !== 0)) {
            let key = this.models.toString().match(/currentView|selectedDate/) || [];
            let propKey = key[0];
            if (eventProp && key && !isUndefined(eventProp[propKey])) {
                if (!isExecute$32) {
                    this.ej2Instances.vueInstance.$emit('update:' + propKey, eventProp[propKey]);
                    this.ej2Instances.vueInstance.$emit('modelchanged', eventProp[propKey]);
                }
                else {
                    this.$emit('update:' + propKey, eventProp[propKey]);
                    this.$emit('modelchanged', eventProp[propKey]);
                }
            }
        }
        if ((this.ej2Instances && this.ej2Instances._trigger)) {
            this.ej2Instances._trigger(eventName, eventProp, successHandler);
        }
    }
    render(createElement) {
        let h = gh || createElement;
        let slots = null;
        if (!isNullOrUndefined(this.$slots.default)) {
            slots = gh ? this.$slots.default() : this.$slots.default;
        }
        return h('div', slots);
    }
    custom() {
        this.updated();
    }
    chartModuleInjection() {
        return this.ej2Instances.chartModuleInjection();
    }
    findCurrentData(totalData, xName) {
        return this.ej2Instances.findCurrentData(totalData, xName);
    }
    rangeChanged(updatedStart, updatedEnd) {
        return this.ej2Instances.rangeChanged(updatedStart, updatedEnd);
    }
    renderPeriodSelector() {
        return this.ej2Instances.renderPeriodSelector();
    }
    stockChartDataManagerSuccess() {
        return this.ej2Instances.stockChartDataManagerSuccess();
    }
};
StockChartComponent = __decorate$32([
    EJComponentDecorator({
        props: properties$5,
        model: {
            event: 'modelchanged'
        }
    }, isExecute$32)
    /* Start Options({
        props: props,
        watch: watch,
        emits: emitProbs,
        provide: function provide() {
            return {
                custom: this.custom
            };
        }
    }) End */
], StockChartComponent);
const StockChartPlugin = {
    name: 'ejs-stockchart',
    install(Vue$$1) {
        Vue$$1.component(StockChartPlugin.name, StockChartComponent);
        Vue$$1.component(StockChartSeriesPlugin.name, StockChartSeriesDirective);
        Vue$$1.component(StockChartSeriesCollectionPlugin.name, StockChartSeriesCollectionDirective);
        Vue$$1.component(StockChartTrendlinePlugin.name, StockChartTrendlineDirective);
        Vue$$1.component(StockChartTrendlinesPlugin.name, StockChartTrendlinesDirective);
        Vue$$1.component(StockChartAxisPlugin.name, StockChartAxisDirective);
        Vue$$1.component(StockChartAxesPlugin.name, StockChartAxesDirective);
        Vue$$1.component(StockChartRowPlugin.name, StockChartRowDirective);
        Vue$$1.component(StockChartRowsPlugin.name, StockChartRowsDirective);
        Vue$$1.component(StockChartAnnotationPlugin.name, StockChartAnnotationDirective);
        Vue$$1.component(StockChartAnnotationsPlugin.name, StockChartAnnotationsDirective);
        Vue$$1.component(StockChartSelectedDataIndexPlugin.name, StockChartSelectedDataIndexDirective);
        Vue$$1.component(StockChartSelectedDataIndexesPlugin.name, StockChartSelectedDataIndexesDirective);
        Vue$$1.component(StockChartPeriodPlugin.name, StockChartPeriodDirective);
        Vue$$1.component(StockChartPeriodsPlugin.name, StockChartPeriodsDirective);
        Vue$$1.component(StockEventPlugin.name, StockEventDirective);
        Vue$$1.component(StockEventsPlugin.name, StockEventsDirective);
        Vue$$1.component(StockChartIndicatorPlugin.name, StockChartIndicatorDirective);
        Vue$$1.component(StockChartIndicatorsPlugin.name, StockChartIndicatorsDirective);
    }
};

var __decorate$33 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const isExecute$33 = gh ? false : true;
let vueImport$27;
if (!isExecute$33 || parseInt(allVue.version) < 3) {
    vueImport$27 = Vue;
}
else {
    vueImport$27 = Vue$1;
}
let BulletRangeCollectionDirective = 
/* Start Options({
    inject: {
        custom: {
            default: null
        }
    }
}) End */
class BulletRangeCollectionDirective extends vueImport$27 {
    constructor() {
        super(arguments);
    }
    render(createElement) {
        if (gh) {
            let h = gh || createElement;
            let slots = null;
            if (!isNullOrUndefined(this.$slots.default)) {
                slots = gh ? this.$slots.default() : this.$slots.default;
            }
            return h('div', { class: 'e-directive' }, slots);
        }
        return;
    }
    updated() {
        if (gh && this.custom) {
            this.custom();
        }
    }
    getTag() {
        return 'e-bullet-range-collection';
    }
};
BulletRangeCollectionDirective = __decorate$33([
    EJComponentDecorator({}, isExecute$33)
    /* Start Options({
        inject: {
            custom: {
                default: null
            }
        }
    }) End */
], BulletRangeCollectionDirective);
const BulletRangeCollectionPlugin = {
    name: 'e-bullet-range-collection',
    install(Vue$$1) {
        Vue$$1.component(BulletRangeCollectionPlugin.name, BulletRangeCollectionDirective);
    }
};
let BulletRangeDirective = class BulletRangeDirective extends vueImport$27 {
    render() {
        return;
    }
    getTag() {
        return 'e-bullet-range';
    }
};
BulletRangeDirective = __decorate$33([
    EJComponentDecorator({}, isExecute$33)
], BulletRangeDirective);
const BulletRangePlugin = {
    name: 'e-bullet-range',
    install(Vue$$1) {
        Vue$$1.component(BulletRangePlugin.name, BulletRangeDirective);
    }
};

var __decorate$34 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
// {{VueImport}}
const properties$6 = ['isLazyUpdate', 'animation', 'border', 'categoryField', 'categoryLabelStyle', 'dataLabel', 'dataSource', 'enableGroupSeparator', 'enablePersistence', 'enableRtl', 'height', 'interval', 'labelFormat', 'labelPosition', 'labelStyle', 'legendSettings', 'locale', 'majorTickLines', 'margin', 'maximum', 'minimum', 'minorTickLines', 'minorTicksPerInterval', 'opposedPosition', 'orientation', 'query', 'ranges', 'subtitle', 'subtitleStyle', 'tabIndex', 'targetColor', 'targetField', 'targetTypes', 'targetWidth', 'theme', 'tickPosition', 'title', 'titlePosition', 'titleStyle', 'tooltip', 'type', 'valueBorder', 'valueField', 'valueFill', 'valueHeight', 'width', 'beforePrint', 'bulletChartMouseClick', 'legendRender', 'load', 'loaded', 'tooltipRender'];
const modelProps$6 = ['dataSource'];
const testProp$6 = getProps({ props: properties$6 });
const props$6 = testProp$6[0];
const watch$6 = testProp$6[1];
const emitProbs$6 = Object.keys(watch$6);
emitProbs$6.push('modelchanged');
for (let props of modelProps$6) {
    emitProbs$6.push('update:' + props);
}
const isExecute$34 = gh ? false : true;
/**
 * Represents Vuejs BulletChart Component
 * ```vue
 * <ejs-bulletchart></ejs-bulletchart>
 * ```
 */
let BulletChartComponent = 
/* Start Options({
    props: props,
    watch: watch,
    emits: emitProbs,
    provide: function provide() {
        return {
            custom: this.custom
        };
    }
}) End */
class BulletChartComponent extends ComponentBase {
    constructor() {
        super(arguments);
        this.propKeys = properties$6;
        this.models = modelProps$6;
        this.hasChildDirective = true;
        this.hasInjectedModules = true;
        this.tagMapper = { "e-bullet-range-collection": "e-bullet-range" };
        this.tagNameMapper = { "e-bullet-range-collection": "e-ranges" };
        this.isVue3 = !isExecute$34;
        this.ej2Instances = new BulletChart({});
        this.ej2Instances._trigger = this.ej2Instances.trigger;
        this.ej2Instances.trigger = this.trigger;
        this.bindProperties();
        this.ej2Instances._setProperties = this.ej2Instances.setProperties;
        this.ej2Instances.setProperties = this.setProperties;
        this.ej2Instances.clearTemplate = this.clearTemplate;
        this.updated = this.updated;
    }
    clearTemplate(templateNames) {
        if (!templateNames) {
            templateNames = Object.keys(this.templateCollection || {});
        }
        if (templateNames.length && this.templateCollection) {
            for (let tempName of templateNames) {
                let elementCollection = this.templateCollection[tempName];
                if (elementCollection && elementCollection.length) {
                    for (let ele of elementCollection) {
                        let destroy = getValue('__vue__.$destroy', ele);
                        if (destroy) {
                            ele.__vue__.$destroy();
                        }
                        if (ele.innerHTML) {
                            ele.innerHTML = '';
                        }
                    }
                    delete this.templateCollection[tempName];
                }
            }
        }
    }
    setProperties(prop, muteOnChange) {
        if (this.isVue3) {
            this.models = !this.models ? this.ej2Instances.referModels : this.models;
        }
        if (this.ej2Instances && this.ej2Instances._setProperties) {
            this.ej2Instances._setProperties(prop, muteOnChange);
        }
        if (prop && this.models && this.models.length) {
            Object.keys(prop).map((key) => {
                this.models.map((model) => {
                    if ((key === model) && !(/datasource/i.test(key))) {
                        if (this.isVue3) {
                            this.ej2Instances.vueInstance.$emit('update:' + key, prop[key]);
                        }
                        else {
                            this.$emit('update:' + key, prop[key]);
                            this.$emit('modelchanged', prop[key]);
                        }
                    }
                });
            });
        }
    }
    trigger(eventName, eventProp, successHandler) {
        if (!isExecute$34) {
            this.models = !this.models ? this.ej2Instances.referModels : this.models;
        }
        if ((eventName === 'change' || eventName === 'input') && this.models && (this.models.length !== 0)) {
            let key = this.models.toString().match(/checked|value/) || [];
            let propKey = key[0];
            if (eventProp && key && !isUndefined(eventProp[propKey])) {
                if (!isExecute$34) {
                    this.ej2Instances.vueInstance.$emit('update:' + propKey, eventProp[propKey]);
                    this.ej2Instances.vueInstance.$emit('modelchanged', eventProp[propKey]);
                }
                else {
                    if (eventName === 'change' || (this.$props && !this.$props.isLazyUpdate)) {
                        this.$emit('update:' + propKey, eventProp[propKey]);
                        this.$emit('modelchanged', eventProp[propKey]);
                    }
                }
            }
        }
        else if ((eventName === 'actionBegin' && eventProp.requestType === 'dateNavigate') && this.models && (this.models.length !== 0)) {
            let key = this.models.toString().match(/currentView|selectedDate/) || [];
            let propKey = key[0];
            if (eventProp && key && !isUndefined(eventProp[propKey])) {
                if (!isExecute$34) {
                    this.ej2Instances.vueInstance.$emit('update:' + propKey, eventProp[propKey]);
                    this.ej2Instances.vueInstance.$emit('modelchanged', eventProp[propKey]);
                }
                else {
                    this.$emit('update:' + propKey, eventProp[propKey]);
                    this.$emit('modelchanged', eventProp[propKey]);
                }
            }
        }
        if ((this.ej2Instances && this.ej2Instances._trigger)) {
            this.ej2Instances._trigger(eventName, eventProp, successHandler);
        }
    }
    render(createElement) {
        let h = gh || createElement;
        let slots = null;
        if (!isNullOrUndefined(this.$slots.default)) {
            slots = gh ? this.$slots.default() : this.$slots.default;
        }
        return h('div', slots);
    }
    custom() {
        this.updated();
    }
    createSvg(chart) {
        return this.ej2Instances.createSvg(chart);
    }
    export(type, fileName, orientation, controls, width, height, isVertical) {
        return this.ej2Instances.export(type, fileName, orientation, controls, width, height, isVertical);
    }
    print(id) {
        return this.ej2Instances.print(id);
    }
    removeSvg() {
        return this.ej2Instances.removeSvg();
    }
};
BulletChartComponent = __decorate$34([
    EJComponentDecorator({
        props: properties$6,
        model: {
            event: 'modelchanged'
        }
    }, isExecute$34)
    /* Start Options({
        props: props,
        watch: watch,
        emits: emitProbs,
        provide: function provide() {
            return {
                custom: this.custom
            };
        }
    }) End */
], BulletChartComponent);
const BulletChartPlugin = {
    name: 'ejs-bulletchart',
    install(Vue$$1) {
        Vue$$1.component(BulletChartPlugin.name, BulletChartComponent);
        Vue$$1.component(BulletRangePlugin.name, BulletRangeDirective);
        Vue$$1.component(BulletRangeCollectionPlugin.name, BulletRangeCollectionDirective);
    }
};

export { TrendlinesDirective, TrendlineDirective, TrendlinesPlugin, TrendlinePlugin, SegmentsDirective, SegmentDirective, SegmentsPlugin, SegmentPlugin, SeriesCollectionDirective, SeriesDirective, SeriesCollectionPlugin, SeriesPlugin, StripLinesDirective, StripLineDirective, StripLinesPlugin, StripLinePlugin, CategoriesDirective, CategoryDirective, CategoriesPlugin, CategoryPlugin, MultiLevelLabelsDirective, MultiLevelLabelDirective, MultiLevelLabelsPlugin, MultiLevelLabelPlugin, AxesDirective, AxisDirective, AxesPlugin, AxisPlugin, RowsDirective, RowDirective, RowsPlugin, RowPlugin, ColumnsDirective, ColumnDirective, ColumnsPlugin, ColumnPlugin, RangeColorSettingsDirective, RangeColorSettingDirective, RangeColorSettingsPlugin, RangeColorSettingPlugin, AnnotationsDirective, AnnotationDirective, AnnotationsPlugin, AnnotationPlugin, SelectedDataIndexesDirective, SelectedDataIndexDirective, SelectedDataIndexesPlugin, SelectedDataIndexPlugin, IndicatorsDirective, IndicatorDirective, IndicatorsPlugin, IndicatorPlugin, ChartComponent, ChartPlugin, AccumulationSeriesCollectionDirective, AccumulationSeriesDirective, AccumulationSeriesCollectionPlugin, AccumulationSeriesPlugin, AccumulationAnnotationsDirective, AccumulationAnnotationDirective, AccumulationAnnotationsPlugin, AccumulationAnnotationPlugin, AccumulationChartComponent, AccumulationChartPlugin, RangenavigatorSeriesCollectionDirective, RangenavigatorSeriesDirective, RangenavigatorSeriesCollectionPlugin, RangenavigatorSeriesPlugin, RangeNavigatorComponent, RangeNavigatorPlugin, RangeBandSettingsDirective, RangeBandSettingDirective, RangeBandSettingsPlugin, RangeBandSettingPlugin, SparklineComponent, SparklinePlugin, SmithchartSeriesCollectionDirective, SmithchartSeriesDirective, SmithchartSeriesCollectionPlugin, SmithchartSeriesPlugin, SmithchartComponent, SmithchartPlugin, StockChartTrendlinesDirective, StockChartTrendlineDirective, StockChartTrendlinesPlugin, StockChartTrendlinePlugin, StockChartSeriesCollectionDirective, StockChartSeriesDirective, StockChartSeriesCollectionPlugin, StockChartSeriesPlugin, StockChartAxesDirective, StockChartAxisDirective, StockChartAxesPlugin, StockChartAxisPlugin, StockChartRowsDirective, StockChartRowDirective, StockChartRowsPlugin, StockChartRowPlugin, StockChartAnnotationsDirective, StockChartAnnotationDirective, StockChartAnnotationsPlugin, StockChartAnnotationPlugin, StockChartSelectedDataIndexesDirective, StockChartSelectedDataIndexDirective, StockChartSelectedDataIndexesPlugin, StockChartSelectedDataIndexPlugin, StockChartPeriodsDirective, StockChartPeriodDirective, StockChartPeriodsPlugin, StockChartPeriodPlugin, StockEventsDirective, StockEventDirective, StockEventsPlugin, StockEventPlugin, StockChartIndicatorsDirective, StockChartIndicatorDirective, StockChartIndicatorsPlugin, StockChartIndicatorPlugin, StockChartComponent, StockChartPlugin, BulletRangeCollectionDirective, BulletRangeDirective, BulletRangeCollectionPlugin, BulletRangePlugin, BulletChartComponent, BulletChartPlugin };
export * from '@syncfusion/ej2-charts';
